demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -94,6 +92,197 @@ def _is_incident_type_default_playbook_found(self, incident_type_data):
+            if not self.handle_error(error_message, error_code, file_path=layouts_container_file_path):
+                is_valid = True
+
+        return is_valid
+
+    def _is_layout_scripts_found(self, layout_data, layout_file_path=None):
+        """Check if scripts of a layout  is in the id_set
+
+        Args:
+            layout_data (dict): Dictionary that holds the extracted details from the given layout.
+            layout_file_path (str): Path to the file.
+
+        Returns:
+            bool. Whether the scripts are in the id_set or not.
+        """
+        is_valid = True
+        scripts_not_in_id_set = set()
+
+        layout_id = list(layout_data.keys())[0]
+        layout = layout_data.get(layout_id, {})
+        layout_name = layout.get('typename', layout_id)
+        scripts = layout.get('scripts', [])
+        scripts_set = set(scripts)
+
+        # Check if the layouts container's scripts are in the id_set:
+        if scripts_set:
+            scripts_not_in_id_set = self._get_scripts_that_are_not_in_id_set(scripts_set)
+
+        # Add error message if there are scripts that aren't in the id_set:
+        if scripts_not_in_id_set:
+            is_valid = False
+            scripts_not_in_id_set_str = ', '.join(scripts_not_in_id_set)
+            error_message, error_code = Errors.layout_non_existent_script_id(layout_name,
+                                                                             scripts_not_in_id_set_str)
+            if not self.handle_error(error_message, error_code, file_path=layout_file_path):
+                is_valid = True
+
+        return is_valid
+
+    def _get_scripts_that_are_not_in_id_set(self, scripts_set):
+        """
+        For each script ID in the given scripts set checks if it is exist in the id set.
+        If a script is in the id set removes it from the input scripts set.
+
+        Args:
+            scripts_set: A set of scripts IDs
+
+        Returns:
+            A sub set of the input scripts set which contains only scripts that are not in the id set.
+        """
+        for checked_script in self.script_set:
+            checked_script_id = list(checked_script.keys())[0]
+            if checked_script_id in scripts_set:
+                scripts_set.remove(checked_script_id)
+
+        # Ignore Builtin scripts because they are implemented on the server side and thus not in the id_set.json
+        scripts_set = self._remove_builtin_scripts(scripts_set)
+
+        # Ignore integration commands scripts because they are not in the id_set.json
+        # Ignoring integration commands is temporary. Validate command should verify that each integration command
+        # called from a layout, a layoutscontainer or an incident field is really exist.
+        # will be fixed in: https://github.com/demisto/etc/issues/41246
+        scripts_set = self._remove_integration_commands_scripts(scripts_set)
```

Comment:
in the issue itself please point to this code and ask for its removal as part of the issue.